site/app/templates/OfficeHoursQueueInstructor.twig

```diff
@@ -0,0 +1,116 @@
+<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
+
+<div class="content">
+  <h1> Office Hours Queue Instructor View </h1> <br/>
+  {% if num_in_queue == 1 %}
+    <h3> There is currently {{num_in_queue}} person in the queue</h3> <br/>
+  {% else %}
+    <h3> There are currently {{num_in_queue}} people in the queue</h3> <br/>
+  {% endif %}
+
+  {% if queue_open == 1 %}
+    <form method="post" id="open_close" action="{{toggle_open_url}}" onsubmit="return confirm('Are you sure you want to close the queue? This will kick everyone out of the queue.');">
+  {% else %}
+    <form method="post" id="open_close" action="{{toggle_open_url}}" onsubmit="return confirm('Are you sure you want to open the queue?');">
+  {% endif %}
+    <input type="hidden" name="csrf_token" value="{{ csrf_token }}" />
+    {% if queue_open == 1 %}
+      <input type="submit" name="queue_close" value="Close Queue" class="btn btn-primary">
+    {% else %}
+      <input type="submit" name="queue_open" value="Open Queue" class="btn btn-primary">
+    {% endif %}
+  </form>
+
+  <br/>
+
+  <h3> Current code: {{ code }} </h3>
+
+  <br/>
+
+  <form method="post" id="get_new_code" action="{{new_code_url}}" onsubmit="return confirm('Are you sure you want to generate a new code?');">
+    <input type="hidden" name="csrf_token" value="{{ csrf_token }}" />
+    <input type="submit" value="Generate New Code" class="btn btn-primary">
+  </form>
+
+  <br/> <br/>
+
+  <table class="table table-striped" style="width:100%;">
+    <thead>
+      <tr>
+        <th scope="col">#</th>
+        <th scope="col">Time</th>
+        <th scope="col">ID</th>
+        <th scope="col">Name</th>
+        <th scope="col">Status</th>
+        <th scope="col">Remove</th>
+      </tr>
+    </thead>
+    <tbody>
+    {% for entry in entries %}
+    <tr>
+      <th scope="row">{{entry.getPositionInQueue() }}</th>
+      <td>{{ entry.getTimeIn() }}</td>
+      <td>{{ entry.getUserId() }}</td>
+      <td>{{ entry.getName() }}</td>
```

Comment:
> It allows students to join the queue with a preferred name for anonymity. 
If the intention is to make it so people can be on the list completely anonymously, why is their identifying user_id also shown in this and the previously helped lists next to their anonymously chosen name?

Reply:
Only the TAs can see the names.    The TAs will call this name out loud in a crowded office hours.  Students could use an alias for privacy reasons, but the system & TAs will know (can look up) the actual student info for tracking purposes.